File: codeforces/training/CF279-D2-B.py
# Codeforces - Books
# https://codeforces.com/contest/279/problem/B
#
# Author: eloyhz
# Date: Oct/22/2021
#

n, t = [int(x) for x in input().split()]
a = [int(x) for x in input().split()]
left, right = 0, 0
ans = 0
s = 0
while right < n:
    s += a[right]
    while s > t and left < right:
        s -= a[left]
        left += 1
    if s <= t:
        ans = max(ans, right - left + 1)
    right += 1
print(ans)

File: codeforces/training/CF279-D2-B.cpp
// Codeforces - Books
// https://codeforces.com/contest/279/problem/B
//
// Author: eloyhz
// Date: Oct/25/2021

#include <bits/stdc++.h>

using namespace std;
using ll = long long;

#define DBG(x) cerr << #x << " = " << (x) << endl
#define DBGY(x) cerr << #x << " = " << (x) << " , "
#define LINE cerr << "================" << endl

int n, t, s;
vector<int> a;

void add(int x) {
    s += x;
}

void remove(int x)  {
    s -= x;
}

bool good()    {
    return s <= t;
}

int main()  {
    cin.tie(nullptr)->sync_with_stdio(false);
    cin >> n >> t;
    a.resize(n);
    for (int &x : a)
        cin >> x;

    int l = 0;
    int ans = 0;
    for (int r = 0; r < n; r++) {
        add(a[r]);
        while (!good()) {
            remove(a[l]);
            l++;
        }
        ans = max(ans, r - l + 1);
    }
    cout << ans << endl;
}
